ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -15,102 +15,58 @@
+import io.confluent.ksql.parser.tree.DropStatement;
 import io.confluent.ksql.serde.DataSource;
-import io.confluent.ksql.services.KafkaTopicClient;
-import io.confluent.ksql.util.ExecutorUtil;
-import io.confluent.ksql.util.KsqlConstants;
 import io.confluent.ksql.util.KsqlException;
-import java.util.Collections;
-import java.util.List;
 
 public class DropSourceCommand implements DdlCommand {
 
   private final String sourceName;
   private final boolean ifExists;
   private final DataSource.DataSourceType dataSourceType;
-  private final KafkaTopicClient kafkaTopicClient;
-  private final SchemaRegistryClient schemaRegistryClient;
   private final boolean deleteTopic;
 
   public DropSourceCommand(
-      final AbstractStreamDropStatement statement,
+      final DropStatement statement,
       final DataSource.DataSourceType dataSourceType,
-      final KafkaTopicClient kafkaTopicClient,
-      final SchemaRegistryClient schemaRegistryClient,
       final boolean deleteTopic) {
 
     this.sourceName = statement.getName().getSuffix();
     this.ifExists = statement.getIfExists();
     this.dataSourceType = dataSourceType;
-    this.kafkaTopicClient = kafkaTopicClient;
-    this.schemaRegistryClient = schemaRegistryClient;
     this.deleteTopic = deleteTopic;
   }
 
   @Override
   public DdlCommandResult run(final MutableMetaStore metaStore) {
     final StructuredDataSource dataSource = metaStore.getSource(sourceName);
+
     if (dataSource == null) {
       if (ifExists) {
         return new DdlCommandResult(true, "Source " + sourceName + " does not exist.");
       }
       throw new KsqlException("Source " + sourceName + " does not exist.");
     }
+
     if (dataSource.getDataSourceType() != dataSourceType) {
       throw new KsqlException(String.format(
           "Incompatible data source type is %s, but statement was DROP %s",
           dataSource.getDataSourceType() == DataSource.DataSourceType.KSTREAM ? "STREAM" : "TABLE",
           dataSourceType == DataSource.DataSourceType.KSTREAM ? "STREAM" : "TABLE"
       ));
     }
-    final DropTopicCommand dropTopicCommand =
-        new DropTopicCommand(dataSource.getKsqlTopicName());
+
+    final DropTopicCommand dropTopicCommand = new DropTopicCommand(dataSource.getKsqlTopicName());
     metaStore.deleteSource(sourceName);
     dropTopicCommand.run(metaStore);
 
-    deleteTopicIfNeeded(dataSource);
-
-    return new DdlCommandResult(
-        true,
-        "Source " + sourceName + " was dropped. "
-            + (deleteTopic ? "Topic '"
-            + dataSource.getKafkaTopicName()
-            + "' was marked for deletion. Actual deletion "
-            + "and removal from brokers may take some time "
-            + "to complete." : ""));
-  }
-
-  private void deleteTopicIfNeeded(final StructuredDataSource dataSource) {
-    if (!deleteTopic) {
-      return;
+    if (deleteTopic) {
+      throw new KsqlException("Should not be executing a delete topic in the engine!");
```

Comment:
If deleteTopic is not needed in this class, isn't it better to remove it from the constructor parameters?

Reply:
Removing this actually fixes the backwards compatibility bug! 🌮